app.js

```diff
@@ -35,6 +35,44 @@ async function hashText(message) {
   return hash384;
 }
 
+function parseContentType(res) {
+  const type = res.headers.get("content-type").split(";");
```

Comment:
What if the passed type is undefined? Please add an early-return for error cases that returns "script"